### ISSUE TITLE
Verify the go releaser works with right version.

### ISSUE THREAD

**Issue by capri-xiyue** [abcxyz/abc#441]
### TL;DR

When I dowoloaded abc cli v0.5.0 via https://github.com/abcxyz/abc/releases/tag/v0.5.0, and then run `abc templates -v`, it showed `abc (devel) (3d99e9bae57b99bc551904ab1b916ecd3c9c4118, darwin/amd64)`, ideally I think it should show `abc 0.5.0 (3d99e9bae57b99bc551904ab1b916ecd3c9c4118, darwin/amd64)` need to check whether https://github.com/abcxyz/abc/pull/439 fix the bug.

When I run `go install github.com/abcxyz/abc/cmd/abc@v0.5.0` and then run `abc templates -v`, it showed `abc v0.5.0 (HEAD, darwin/amd64)`. I think this part it works correctly

cc @drevell 

### Environment Details

```markdown
v0.0.5
```

**Comment by sethvargo** [comment]
Yea, sorry about this. I introduced that bug with the `versioninfo` package, and I thought I had fixed it everywhere.

**Comment by capri-xiyue** [comment]
> Yea, sorry about this. I introduced that bug with the `versioninfo` package, and I thought I had fixed it everywhere.

Thanks for fixing it. With current mechanism, I see some inconsistency. Not sure whether it's a common practise or not.

For example,

I think currently when I install via 
1. `go install github.com/abcxyz/abc/cmd/abc@v0.6.0`
- `abc templates -v` will return `abc v0.6.0 (HEAD, darwin/amd64)`
- `go versiom -m <Binary>` will return `mod github.com/abcxyz/abc	v0.6.0`

2. Download v0.6.0 artifacts in GitHub
- `abc templates -v` will return `abc 0.6.0 (<commit-sha>, darwin/amd64), Note: you can see the difference of having `v` prefix or not
- `go versiom -m <Binary>` will return `mod github.com/abcxyz/abc (devel)`, Note: you can see the build info diff as in go releaser, we don't override build info.

**Comment by sethvargo** [comment]
The `v` prefix is weird, but the commit-sha vs `HEAD` is expected

**Comment by capri-xiyue** [comment]
> The `v` prefix is weird, but the commit-sha vs `HEAD` is expected

I guess the `v` prefix is from `go install github.com/abcxyz/abc/cmd/abc@v0.6.0`, I think by default the `go build` involved in `go install` will use the `v0.6.0` tag.

**Comment by drevell** [comment]
Verified fix:

```
ct2:~:$ go install github.com/abcxyz/abc/cmd/abc@v0.6.0-rc1
ct2:~:$ abc -version
abc v0.6.0-rc1 (HEAD, linux/amd64)
```
